presto-main-base/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -1437,8 +1437,32 @@ else if (character == '%') {
     @SqlType(StandardTypes.INTERVAL_DAY_TO_SECOND)
     public static long parseDuration(@SqlType("varchar(x)") Slice duration)
     {
+        String durationStr = duration.toStringUtf8();
```

Comment:
```suggestion
        String durationStr = duration.toStringUtf8().trim();
```

Forgot to handle the trailing spaces. The calculation for something like `parse_duration('7702741401940153   ms  ')` will fall back.